libazureinit/src/provision/ssh.rs

```diff
@@ -250,10 +250,11 @@ pub(crate) fn update_sshd_config(
         temp_file.write_all(modified_content.as_bytes())?;
         temp_file.set_permissions(fs::Permissions::from_mode(0o644))?;
 
-        fs::rename(temp_sshd_config_path, &sshd_config_path)?;
+        fs::copy(temp_sshd_config_path, &sshd_config_path)?;
+        fs::remove_file(temp_sshd_config_path)?;
         tracing::info!(
             ?sshd_config_path,
-            "Updated existing sshd setting to allow password authentication"
+            "Updated existing sshd setting to allow password authentication (Mariner)"
```

Comment:
I'd recommend leaving the log statement as-is since it could easily _not_ be Mariner/Azure Linux; this path gets hit anytime the file exists (the config snippet could exist from a previous provisioning, for example)